examples/src/main/java/kafka/examples/Consumer.java

```diff
@@ -157,9 +157,4 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
     public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         Utils.printOut("Assigned partitions: %s", partitions);
     }
-
-    @Override
-    public void onPartitionsLost(Collection<TopicPartition> partitions) {
-        Utils.printOut("Lost partitions: %s", partitions);
```

Comment:
I don't really understand the motivation of this change. It seems to me that we are losing the log if we do so. Could you elaborate more on the motivation?

Reply:
Hi @dajac, the default implementation of onPartitionsLost calls onPartitionsRevoked so that any cleanup logic can be executed (i.e. commit pending offsets before losing the partition ownership). This also means that "revoked" is logged instead of "lost".

https://github.com/fvaleri/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.java#L197-L199

For this readon, I thought we could simply use the default implementation without overriding, but now I see that it is too late to save the offsets when onPartitionsLost is called, since these partitions are probably owned by other consumers already.

From ConsumerRebalanceListener javadoc:

```sh
public void onPartitionsLost(Collection<TopicPartition> partitions) {
    // do not need to save the offsets since these partitions are probably owned by other consumers already
}
```

I guess the default onPartitionsLost implementation is there to cover some corner case where onPartitionsRevoked may not be triggered, but we may still need to do some other cleanup. Is that correct? Maybe we can leave it adding an appropriate comment. Wdyt?

Reply:
I am sorry but I don't understand why you are trying to achieve here. Your explanation is correct and the javadoc of `onPartitionsLost` also explains this. I don't see the need for adding an extra comment just in this method in this example. Is your goal trying to improve the example?

Reply:
Yeah, we refactored these examples a while back, keeping the original logic, but improving error handling and output. We also added some comments, which may be useful to people new to Kafka. Adding this comment may be along these lines.

Reply:
In my opinion, if you want to do something like that, it would be better to do it for all callbacks then. Adding just one comment there is not an ideal end state.

Reply:
Right. Let me do that.